lib/CGI/Application.pm

```diff
@@ -1078,12 +1078,36 @@ support to it.
 The simplest way to create and return a PSGI-compatible coderef. Pass in
 arguments to a hashref just as would to new. This returns a PSGI-compatible
 coderef, using L<CGI:::PSGI> as the query object. To use a different query
-object, construct your own object using C<< run_as_psgi() >>, as shown below.
+object, you have to construct your own object using C<< run_as_psgi() >>, 
+as shown below, because C<< psgi_app >> creates a new CGI::PSGI object
+unconditionally on every request and therefore overrides a own query object.
 
 It's possible that we'll change from CGI::PSGI to a different-but-compatible
 query object for PSGI support in the future, perhaps if CGI.pm adds native
 PSGI support.
 
+The use of psgi_app might look like this:
+
+	package WebApp;
+	use base "CGI::Application";
+
+	sub setup {
+		my $self = shift;
+		$self->start_mode('welcome');
+		$self->mode_param('rm');
+		$self->run_modes(
+		          'welcome' => 'welcome'
+		  );
+	}
+
+	sub welcome {
+		return "Hello World";
+	}
+
+	package main;
+
+	my $app = WebApp->psgi_app();
+
```

Comment:
I don't think that there's a need for this example here. The example I was talking about at https://github.com/MartinMcGrath/CGI--Application/issues/7 was essentially the one above: line
` $psgi_coderef = WebApp->psgi_app({ ... args to new() ... });` - should be changed to something like 
` $psgi_coderef = WebApp->psgi_app({ ... args to new() without QUERY... });` - that's it.